lua/yabs/init.lua

```diff
@@ -73,20 +74,37 @@ function Yabs:add_task(name, args)
 end
 
 function Yabs:get_current_language_tasks()
+    if not self.did_setup then return {} end
     return self:get_current_language().tasks
 end
 
 function Yabs:get_global_tasks()
     return self.tasks
 end
 
-function Yabs:get_tasks()
-    local current_tasks, global_tasks = self:get_current_language_tasks(), self:get_global_tasks()
-    local tasks = vim.tbl_extend("keep", current_tasks, global_tasks)
+function Yabs:get_tasks(scope)
+    local local_tasks, global_tasks = self:get_current_language_tasks(), self:get_global_tasks()
+    if scope then
+        if scope == scopes.GLOBAL then
+            return local_tasks
+        end
+        if scope == scopes.LOCAL then
+            return global_tasks
+        end
+    end
+    local tasks = vim.tbl_extend("keep", local_tasks, global_tasks)
```

Comment:
Is it just me or conditions are reverted? For global tasks you return local tasks and vice versa.

Also maybe lazy-evaluate this part?

```suggestion
    if scope then
        if scope == scopes.GLOBAL then
            return self:get_global_tasks()
        end
        if scope == scopes.LOCAL then
            return self:get_current_language_tasks()
        end
    end
    local tasks = vim.tbl_extend("keep", self:get_current_language_tasks(), self:get_global_tasks())
```

Reply:
Lol, you’re right, will fix this.
Thanks!